extensions/positron-python/src/test/mocks/helper.ts

```diff
@@ -1,11 +1,30 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-
+import * as TypeMoq from 'typemoq';
 import { Readable } from 'stream';
+// eslint-disable-next-line import/no-unresolved
+import * as common from 'typemoq/Common/_all';
 
 export class FakeReadableStream extends Readable {
     _read(_size: unknown): void | null {
         // custom reading logic here
         this.push(null); // end the stream
     }
 }
+
+export function createTypeMoq<T>(
+    targetCtor?: common.CtorWithArgs<T>,
+    behavior?: TypeMoq.MockBehavior,
+    shouldOverrideTarget?: boolean,
+    ...targetCtorArgs: any[]
+): TypeMoq.IMock<T> {
+    // Use typemoqs for those things that are resolved as promises. mockito doesn't allow nesting of mocks. ES6 Proxy class
+    // is the problem. We still need to make it thenable though. See this issue: https://github.com/florinn/typemoq/issues/67
+    const result = TypeMoq.Mock.ofType<T>(targetCtor, behavior, shouldOverrideTarget, ...targetCtorArgs);
+    // --- Start Positron ---
+    // result.setup((x: any) => x.then).returns(() => undefined);
```

Comment:
this line caused every single usage of this function to fail with `TypeError: Cannot convert a Symbol value to a string`-- I'm not sure why since it seems to be a community vetted way to mock things resolved as promises and works upstream 😩

Reply:
Looks like we're running into a version of https://github.com/inversify/InversifyJS/issues/1510 based on the error output, but it is odd there seemingly aren't issues with it upstream 🤷 

```
1) Interpreters service
       Changes to active document should invoke interpreter.refresh method:
     TypeError: Cannot convert a Symbol value to a string
      at Object.LAZY_IN_SYNC (node_modules/inversify/lib/constants/error_msgs.js:25:80)
      at Container._getButThrowIfAsync (node_modules/inversify/lib/container/container.js:601:40)
      at Container.get (node_modules/inversify/lib/container/container.js:374:21)
      at ServiceContainer.get (out/client/ioc/container.js:27:89)
      at new InterpreterService (out/client/interpreter/interpreterService.js:86:52)
      at Context.<anonymous> (out/test/interpreters/interpreterService.unit.test.js:142:25)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)
```

Reply:
I ran into this when we upgraded to TS v5 previously. We have some unfortunate hacks in place that helped back then. See https://github.com/posit-dev/positron/issues/2271#issuecomment-1952404450.

Reply:
Ah, I just noticed that they've also upgraded to TS v5 and have implemented their own solution to the above problem. We should probably revert our own workaround in `src/test/vscode-mock.ts` (original PR: https://github.com/posit-dev/positron-python/pull/365). I'm happy to look into this!

Reply:
Yeah, a lot of this update is the TS 4.5.5 -> 5.2 updates. I'll open up a separate issue to revert the workaround and assign it to you if that works!